src/utils.zig

```diff
@@ -0,0 +1,41 @@
+const std = @import("std");
+const assert = std.debug.assert;
+
+pub inline fn div_ceil(numerator: anytype, denominator: anytype) @TypeOf(numerator, denominator) {
+    comptime {
+        const T = @TypeOf(numerator, denominator);
```

Comment:
Could we make this less compound and rather check numerator and denominator unsignedness separately to be crystal clear?